autofit/graphical/expectation_propagation/optimiser.py

```diff
@@ -188,15 +186,25 @@ def __init__(
                 for factor in self.factors
             }
 
-        for optimiser in self.factor_optimisers.values():
-            optimiser.paths = self.paths
-
+            
         self.ep_history = ep_history or EPHistory()
 
-        with open(self.output_path / "graph.info", "w+") as f:
-            f.write(self.factor_graph.info)
+        self.visualiser = None
+        if paths is None:
+            try:
+                paths = DirectoryPaths(identifier=str(Identifier(factor_graph)))
+            except KeyError: 
+                pass 
+
+        self.paths = paths
+        if self.paths:
+            for optimiser in self.factor_optimisers.values():
+                optimiser.paths = self.paths
```

Comment:
We probably want each factor_optimiser to have a different paths to ensure that they do not clash.

If multiple DynestyStatic had the same paths, for example, we might see them loading each other's optimisation state.

You could use the create_child method and pass the name of each factor.

Reply:
I have no idea what `paths` are supposed to do but it was breaking some of my code, so I put in code paths so that it became optional. In a later commit I've set the paths attribute to be optionally set if it doesn't exist.

Reply:
Actually had to revert the change, if you don't have this line `test_autofit/graphical/info/test_output.py` fails.

Reply:
It's basically an object that handles saving and loading data for optimisations. We used to have path manipulation stuff litered all over the optimiser classes so we put it all in one place. It can also conveniently be swapped out for a database client so data is persisted directly into a SQL database rather than being saved in a directory structure.